runtime/src/accounts_db.rs

```diff
@@ -4368,4 +4487,79 @@ pub mod tests {
             shrink_thread.join().unwrap();
         }
     }
+
+    #[test]
+    fn test_account_balance_for_capitalization_normal() {
+        // system accounts
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(10, &Pubkey::default(), false),
+            10
+        );
+        // any random program data accounts
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(10, &Pubkey::new_rand(), false),
+            10
+        );
+    }
+
+    #[test]
+    fn test_account_balance_for_capitalization_sysvar() {
+        use solana_sdk::sysvar::Sysvar;
+
+        let normal_sysvar = solana_sdk::slot_history::SlotHistory::default().create_account(1);
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(
+                normal_sysvar.lamports,
+                &normal_sysvar.owner,
+                normal_sysvar.executable
+            ),
+            0
+        );
+
+        // currently transactions can send any lamports to sysvars although this is not sensible.
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(10, &solana_sdk::sysvar::id(), false),
+            9
+        );
+    }
+
+    #[test]
+    fn test_account_balance_for_capitalization_native_program() {
+        let normal_native_program = solana_sdk::native_loader::create_loadable_account("foo");
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(
+                normal_native_program.lamports,
+                &normal_native_program.owner,
+                normal_native_program.executable
+            ),
+            0
+        );
+
+        // test maliciously assigned bogus native loader account
+        assert_eq!(
+            AccountsDB::account_balance_for_capitalization(
+                1,
+                &solana_sdk::native_loader::id(),
+                false
+            ),
+            1
+        )
+    }
+
+    #[test]
+    fn test_checked_sum_for_capitalization_normal() {
+        assert_eq!(
+            AccountsDB::checked_sum_for_capitalization(vec![1, 2].into_iter()),
+            3
+        );
+    }
+
```

Comment:
Should we have a test (probably in bank.rs) which does some rent and inflation and then checks the capitalization?

Reply:
Yeah, I've updated to verify capitalization in those relevant tests: 95cb371ecd6ba8dfb0527ee4e828081c1d0e937f

Reply:
awesome